### PR TITLE
[heft-sass-plugin] Upgrade postcss-modules

### PR DESCRIPTION
**Pull request by kamontat**

### CODE REVIEW

**Comment by iclanton** [comment]
What testing have you done on this? What changed in `postcss-modules` in those five major bumps?

File: common/changes/@rushstack/heft-sass-plugin/feature-upgrade-postcss-modules_2023-08-23-11-43.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@rushstack/heft-sass-plugin",
+      "comment": "Upgrade postcss-modules from v1.5.0 to v6.0.0",
+      "type": "patch"
```
Comment (iclanton) [inline common/changes/@rushstack/heft-sass-plugin/feature-upgrade-postcss-modules_2023-08-23-11-43.json:6]:
```suggestion
      "type": "minor"
```

**Comment by kamontat** [comment]
@iclanton Below are the changes collected from [CHANGELOG](https://github.com/madyankin/postcss-modules/blob/master/CHANGELOG.md)

- For [v2.0.0][v2.0.0]: remove `camelCase` key which we didn't use.
- For [v3.0.0][v3.0.0]: dropped Node v8 and v9 support.
- For [v4.0.0][v4.0.0]: upgraded to PostCSS 8 which we already use.
- For [v5.0.0][v5.0.0]: upgrade babel to v7.
- For [v6.0.0][v6.0.0]: change function `resolve` signature which we didn't use.

Sorry for delay, I didn't check my GitHub account notification for some times.

[v2.0.0]: https://github.com/madyankin/postcss-modules/compare/v1.5.0...v2.0.0
[v3.0.0]: https://github.com/madyankin/postcss-modules/compare/v2.0.0...v3.0.0
[v4.0.0]: https://github.com/madyankin/postcss-modules/compare/v3.0.0...v4.0.0
[v5.0.0]: https://github.com/madyankin/postcss-modules/compare/v4.0.0...v5.0.0
[v6.0.0]: https://github.com/madyankin/postcss-modules/compare/v5.0.0...v6.0.0

**Comment by kamontat** [comment]
Sync with main branch.

**Comment by octogonz** [comment]
@iclanton is this PR ready to merge?

**Pull request merged**